### ISSUE TITLE
Mac OS X, packaged app error ((com.hackety.shoes.test... Service exited with abnormal code: 1)

### ISSUE THREAD

**Issue by thebenjy** [shoes/shoes4#1191]
our Shoes3 app needed some tweaking to get working via command line on Shoes4. (had some bugs that were some reason working in shoes3). now working invoked from cmd line.
Attempted to package app specifying file, opening app via Finder bounces then crashes (log below from console).

mac os x, yosemite. 

repeated with ultra basic test app (also attached)

please let me know what other info i can provide. just currently trying to package in windows to see if same issue.

**package**
bin/shoes -p swt:app /Users/ben.forrest/Documents/repositories/savemail/tmp_savemail_test_dir/test_sh4.rb

**simple app**
# simpleshoe.rb

require 'shoes'
$ini_file_lb=nil

Shoes.app(title: "title", width: 900, height: 900) do
stack :width => "98%", margin: 0 do
    background white
    flow do
      image "1.png", :margin => 0
      title "Environment"
      flow do
           $env_r_staging = radio :server, checked: true
           para "Staging (Test)       "
           $env_r_production = radio :server
           para "Production"
         end
    end #flow
  end # stack
end # app

**Log**
7/12/2015 7:37:47.946 pm JavaAppLauncher[33159]: WARNING: The Gestalt selector gestaltSystemVersion is returning 10.9.4 instead of 10.10.4. Use NSProcessInfo's operatingSystemVersion property to get correct system version number.
Call location:
7/12/2015 7:37:47.946 pm JavaAppLauncher[33159]: 0   CarbonCore                          0x00007fff8c3a72b7 ___Gestalt_SystemVersion_block_invoke + 113
7/12/2015 7:37:47.946 pm JavaAppLauncher[33159]: 1   libdispatch.dylib                   0x00007fff875c1c13 _dispatch_client_callout + 8
7/12/2015 7:37:47.946 pm JavaAppLauncher[33159]: 2   libdispatch.dylib                   0x00007fff875c1b26 dispatch_once_f + 117
7/12/2015 7:37:47.946 pm JavaAppLauncher[33159]: 3   CarbonCore                          0x00007fff8c330456 _Gestalt_SystemVersion + 987
7/12/2015 7:37:47.946 pm JavaAppLauncher[33159]: 4   CarbonCore                          0x00007fff8c32f6e3 Gestalt + 144
7/12/2015 7:37:47.947 pm JavaAppLauncher[33159]: 5   libswt-pi-cocoa-4427.jnilib         0x0000000121e581aa Java_org_eclipse_swt_internal_cocoa_OS_Gestalt + 98
7/12/2015 7:37:47.947 pm JavaAppLauncher[33159]: 6   ???                                 0x0000000103e83954 0x0 + 4360517972
7/12/2015 7:37:49.695 pm com.apple.xpc.launchd[1]: (com.hackety.shoes.test_sh4.256484[33159]) Service exited with abnormal code: 1

**Comment by jasonrclark** [comment]
Hey @thebenjy thanks for the bug report! Packaging is in its early days with Shoes 4. It's one of the big remaining pieces before we reach a release candidate, but sorry that it's behaving badly for you!

If you package the app as a jar with the `-p swt:jar` option and run it via `java -jar -XstartOnFirstThread simpleshoe.jar` do you get a different result?

**Comment by PragTob** [comment]
@thebenjy thanks for the report! By any chance, do you remember the bugs you had when converting? Ultimate goal is to run Shoes3 apps out of the box (there are some differences though) and sheos4 still has some bugs so it might as well have been our bugs :o

**Comment by thebenjy** [comment]
@PragTob have a bunch of commits during migration. will raise a separate issue
@jasonrclark i've realised i was using wrong ruby's - let me fix first and revert. I have noticed that shoes4 doesn't seem to be properly installing during rvm usage (i.e create gemset, bundle install, when i list gems shoes-core / shoes not listed:

shoes-highlighter (1.0.1)
shoes-manual (4.0.0)
swt (4.4)

is this a known issue?

**Comment by jasonrclark** [comment]
@thebenjy I haven't see that before but I don't use RVM. Could it have to do with referencing it by `:git` in your `Gemfile`?

**Comment by thebenjy** [comment]
@jasonrclark i'd assumed i'd needed to reference to download the relevant pre version?

**Comment by jasonrclark** [comment]
To use things that are still just on master on Github, that's the simplest option. But we do have pre-release versions of the gem available, so you should be able to specify like this:

```
gem 'shoes', '4.0.0.pre4'
```

**Comment by thebenjy** [comment]
@jasonrclark  it seems to work with shoes installed separately. e.g. samples run. 
packaging the jar file, i get /bin/sh: -c line 0: syntax error when running.

of note the JRUBY_OPTS='-J-XstartOnFirstThread' doesn't seem to be picked up

      $ java -jar pkg/test_sh4.jar 
      Ignoring gem-wrappers-1.2.7 because its extensions are not built.  Try: gem pristine gem-wrappers --version 1.2.7
      ***WARNING: Display must be created on main thread due to Cocoa restrictions.
      Ooops, we couldn't start properly. We'll try again.

      To avoid this restarting behavior, try one of the following:
        * Use the shoes executable to start your app
        * Add JRUBY_OPTS='-J-XstartOnFirstThread' to your environment before starting.
      /bin/sh: -c: line 0: syntax error near unexpected token `newline'
      /bin/sh: -c: line 0: `JRUBY_OPTS=-J-XstartOnFirstThread <script>'
      ben.forrest in ~/Documents/repositories/project/tmp_project_test_dir on master*
      $ export JRUBY_OPTS='-J-XstartOnFirstThread'
      You have new mail in /var/mail/ben.forrest
      ben.forrest in ~/Documents/repositories/project/tmp_project_test_dir on master*
      $ java -jar pkg/test_sh4.jar 
      Ignoring gem-wrappers-1.2.7 because its extensions are not built.  Try: gem pristine gem-wrappers --version 1.2.7
      ***WARNING: Display must be created on main thread due to Cocoa restrictions.
      Ooops, we couldn't start properly. We'll try again.

      To avoid this restarting behavior, try one of the following:
        * Use the shoes executable to start your app
        * Add JRUBY_OPTS='-J-XstartOnFirstThread' to your environment before starting.
      /bin/sh: -c: line 0: syntax error near unexpected token `newline'
      /bin/sh: -c: line 0: `JRUBY_OPTS=-J-XstartOnFirstThread <script>'
      ben.forrest in ~/Documents/repositories/project/tmp_project_test_dir on master*
      $

**Comment by jasonrclark** [comment]
If you're just running `java -jar`, it doesn't see the `JRUBY_OPTS`, so you just need to pass it as `-XstartOnFirstThread` (no preceding `-J`, which is JRuby's way of shuttling along options to the JVM).

**Comment by thebenjy** [comment]
still having issues with rvm - let me raise a separate issue .i think its related tothat problem for now - can't find gems etc.

**Comment by PragTob** [comment]
@thebenjy rvm should be fine, I'm using rvm all the time, on Linux though. Will have a look at the other issue

**Comment by thebenjy** [comment]
well thats good to know. do you use shoes as a gem in gemfile?

**Comment by PragTob** [comment]
Nope, I usually just run from source :)

**Comment by jasonrclark** [comment]
Hey @thebenjy. Guessing from other issues that you might not have time or interest, but I think a reasonable next step here given updates to packaging and JRuby ecosystem around it would be to retry things with the latest `pre9` gem release.

Marking this as needs feedback, and if we don't get around to checking during `pre10` I'll likely close out.

**Comment by thebenjy** [comment]
Apologies. This has fallen off our demand list for now. Client app being used is sufficient . Haven't had time to continue to investigate but appreciate interest and feedback
